python/pyarrow/tests/test_csv.py

```diff
@@ -200,6 +208,30 @@ def test_simple_nulls(self):
             'e': [b"3", b"nan", b"\xff"],
             }
 
+    def test_column_types(self):
+        # Ask for specific column types in ConvertOptions
+        opts = ConvertOptions(column_types={'b': pa.float32(),
+                                            'c': pa.string(),
+                                            'd': pa.null()})
```

Comment:
If I already have a `pa.Schema` object which defines the column types can I pass that directly to `pa.csv.read_csv`?

Reply:
No. Perhaps we want to allow passing a schema as `column_types`? @wesm

Reply:
From a user api point-of-view I think that would be easier than requiring a mapping:
```python
column_types = {field.name: field.type for field in schema}
```
i.e. the schema should be the one way to describe arrow Tables.

A mapping api is made a bit more cumbersome because there's no trivial way to conver to to a schema (I think that would a useful function/method to have):
```python
schema = pa.schema(list(starmap(pa.field, column_types.items())))
```

Reply:
I think it would be good to also accept a Schema, it's another way to indicate the desired column types

Reply:
Some CSV files have hundreds of columns. Take a look at the API for this in the R readr package https://cran.r-project.org/web/packages/readr/vignettes/readr.html

Reply:
> *We will require configurability for some types beyond just a data type eventually (e.g. categorical with known categories)*

I'm not sure I follow - isn't a categorical type an actual data type which is defined by its categories?

Reply:
Sorry, bad example. Let me try again: dates with a particular format (like "%d-%m-%y")

Reply:
Yeah, this (applying a known schema) and date handling are the most important features for me! :)